translate c test fails on windows CI

The "zig compile server - translate c" test in `tests/language_features/cimport.zig` [fails on the windows CI](https://github.com/zigtools/zls/actions/runs/5349008628/jobs/9700501298) with:

```
Run zig build test
  zig build test
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
run test: error: while executing test 'test.zig compile server - translate c', the following command exited with code 9 (expected exited with code 0):
D:\a\zls\zls\zig-cache\o\61a46b4cc[2](https://github.com/zigtools/zls/actions/runs/5349008628/jobs/9700501298#step:8:2)a86277072d6[3](https://github.com/zigtools/zls/actions/runs/5349008628/jobs/9700501298#step:8:3)c638e5f7[4](https://github.com/zigtools/zls/actions/runs/5349008628/jobs/9700501298#step:8:5)c\test.exe --listen=- 
Build Summary: 7/9 steps succeeded; 1 failed; 163/167 tests passed; 4 skipped (disable with --summary none)
test transitive failure
+- run test failure
error: the following build command failed with exit code 1:
D:\a\zls\zls\zig-cache\o\c[5](https://github.com/zigtools/zls/actions/runs/5349008628/jobs/9700501298#step:8:6)[6](https://github.com/zigtools/zls/actions/runs/5349008628/jobs/9700501298#step:8:7)b2[7](https://github.com/zigtools/zls/actions/runs/5349008628/jobs/9700501298#step:8:8)a1e5[8](https://github.com/zigtools/zls/actions/runs/5349008628/jobs/9700501298#step:8:9)26ae5e735bf8654e8f601\build.exe C:\hostedtoolcache\windows\zig\master\x64\zig.exe D:\a\zls\zls D:\a\zls\zls\zig-cache C:\Users\runneradmin\AppData\Local\zig test
Error: Process completed with exit code 1.
```

--- a/src/ZigCompileServer.zig
+++ b/src/ZigCompileServer.zig
@@ -32,32 +32,27 @@ pub fn receiveMessage(client: *Client) !InMessage.Header {
     const Header = InMessage.Header;
     const fifo = client.pooler.fifo(.in);
 
-    while (try client.pooler.poll()) {
-        const buf = fifo.readableSlice(0);
-        assert(fifo.readableLength() == buf.len);
-        if (buf.len >= @sizeOf(Header)) {
-            // workaround for https://github.com/ziglang/zig/issues/14904
+    var first_run = true;
+    var header: ?Header = null;
+    while (first_run or try client.pooler.poll()) {
+        first_run = false;
+
+        if (header == null) {
+            if (fifo.readableLength() < @sizeOf(Header)) continue;
+            const buf = fifo.readableSlice(0);
             const bytes_len = bswap_and_workaround_u32(buf[4..][0..4]);
             const tag = bswap_and_workaround_tag(buf[0..][0..4]);
-
-            if (buf.len - @sizeOf(Header) >= bytes_len) {
-                fifo.discard(@sizeOf(Header));
-                return .{
-                    .tag = tag,
-                    .bytes_len = bytes_len,
-                };
-            } else {
-                const needed = bytes_len - (buf.len - @sizeOf(Header));
-                const write_buffer = try fifo.writableWithSize(needed);
-                const amt = try client.in.readAll(write_buffer);
-                fifo.update(amt);
-                continue;
-            }
+            header = Header{
+                .tag = tag,
+                .bytes_len = bytes_len,
+            };
+            fifo.discard(@sizeOf(Header));
         }
 
-        const write_buffer = try fifo.writableWithSize(256);
-        const amt = try client.in.read(write_buffer);
-        fifo.update(amt);
+        if (header) |h| {
+            if (fifo.readableLength() < h.bytes_len) continue;
+            return h;
+        }
     }
     return error.Timeout;
 }

--- a/src/translate_c.zig
+++ b/src/translate_c.zig
@@ -236,12 +236,10 @@ pub fn translate(
                 const body_size = @sizeOf(std.zig.Server.Message.EmitBinPath);
                 if (header.bytes_len <= body_size) return error.InvalidResponse;
 
-                const trailing_size = header.bytes_len - body_size;
-
                 _ = try zcs.receiveEmitBinPath();
 
-                const result_path = try zcs.receiveBytes(allocator, trailing_size);
-                defer allocator.free(result_path);
+                const trailing_size = header.bytes_len - body_size;
+                const result_path = zcs.pooler.fifo(.in).readableSliceOfLen(trailing_size);
 
                 return Result{ .success = try URI.fromPath(allocator, std.mem.sliceTo(result_path, '\n')) };
             },

--- a/tests/language_features/cimport.zig
+++ b/tests/language_features/cimport.zig
@@ -9,9 +9,6 @@ const translate_c = zls.translate_c;
 const allocator: std.mem.Allocator = std.testing.allocator;
 
 test "zig compile server - translate c" {
-    // FIXME: Disabled due to https://github.com/zigtools/zls/issues/1252
-    if (@import("builtin").os.tag == .windows) return error.SkipZigTest;
-
     var result1 = try testTranslate(
         \\void foo(int);
         \\void bar(float*);

